src/unix/fs.c

```diff
@@ -766,6 +770,136 @@ static ssize_t uv__fs_write(uv_fs_t* req) {
   return r;
 }
 
+static ssize_t uv__fs_copyfile(uv_fs_t* req) {
+#if defined(__APPLE__)
+  /* On macOS, use the native copyfile(3). */
+  copyfile_flags_t flags;
+
+  flags = COPYFILE_ALL;
+
+  if (req->flags & UV_FS_COPYFILE_EXCL)
+    flags = flags | COPYFILE_EXCL;
+
+  return copyfile(req->path, req->new_path, NULL, flags);
+#else
+  uv_fs_t fs_req;
+  uv_file src;
+  uv_file dst;
+  uv_buf_t buf;
+  char data[4096];
+  struct stat statsbuf;
+  int offset;
+  int num_read;
+  int num_written;
+  int dst_flags;
+  int result;
+  int err;
+
+  dst = -1;
+
+  /* Open the source file. */
+  src = uv_fs_open(req->loop, &fs_req, req->path, O_RDONLY, 0, NULL);
+  uv_fs_req_cleanup(&fs_req);
+
+  if (src < 0) {
+    err = src;
+    goto out;
+  }
+
+  /* Get the source file's mode. */
+  err = fstat(src, &statsbuf);
+
+  if (err != 0) {
+    err = -errno;
+    goto out;
+  }
+
+  dst_flags = O_WRONLY | O_CREAT;
+
+  if (req->flags & UV_FS_COPYFILE_EXCL)
+    dst_flags = dst_flags | O_EXCL;
+
+  /* Open the destination file. */
+  dst = uv_fs_open(req->loop,
+                   &fs_req,
+                   req->new_path,
+                   dst_flags,
+                   statsbuf.st_mode,
+                   NULL);
+  uv_fs_req_cleanup(&fs_req);
+
+  if (dst < 0) {
+    err = dst;
+    goto out;
+  }
+
+  /* Set up the buffer and read from the source file in a loop. */
```

Comment:
Yes, but our `uv_fs_sendfile` will take care of that :-) So if `sendfile(2)` works with files on your system it will use that, or fallback to the emulation, since it does runtime detection based on the errno.